SCORER.md

```diff
@@ -21,11 +25,231 @@ You may need to build your own scorer - your own _language model_ if:
+
+4. Downloading the prebuilt `native_client` from the DeepSpeech repository on GitHub, and using the `generate_scorer_package` to create a `kenlm.scorer` file.
+
+5. Using the `kenlm.scorer` file as the _external_scorer_ passed to `DeepSpeech.py` during training.
+
+In the following example we will create a custom external scorer file for Bahasa Indonesia (BCP47: `id-ID`).
+
+#### Preparing the text file
+
+This is straightforward. In this example, we will use a file called `indonesian-sentences.txt`. This file has been created by copying phrases from the `train.tsv`, `validate.tsv` and `test.tsv` files in the Common Voice Indonesian dataset.
+
+```
+~/deepspeech-data$ ls cv-corpus-6.1-2020-12-11/id
+total 6288
+   4 drwxr-xr-x 3 root root    4096 Feb 24 19:01 ./
+   4 drwxr-xr-x 4 root root    4096 Feb 11 07:09 ../
+1600 drwxr-xr-x 2 root root 1638400 Feb  9 10:43 clips/
+ 396 -rwxr-xr-x 1 root root  401601 Feb  9 10:43 dev.tsv
+ 104 -rwxr-xr-x 1 root root  103332 Feb  9 10:43 invalidated.tsv
+1448 -rwxr-xr-x 1 root root 1481571 Feb  9 10:43 other.tsv
+  28 -rwxr-xr-x 1 root root   26394 Feb  9 10:43 reported.tsv
+ 392 -rwxr-xr-x 1 root root  399790 Feb  9 10:43 test.tsv
+ 456 -rwxr-xr-x 1 root root  465258 Feb  9 10:43 train.tsv
+1848 -rwxr-xr-x 1 root root 1889606 Feb  9 10:43 validated.tsv
+```
+
+The `indonesian-sentences.txt` file is stored on the local filesystem in the `deepspeech-data` directory so that the Docker container can access it.
+
+```
+~/deepspeech-data$ ls | grep indonesian-sentences
+ 476 -rw-rw-r--  1 root root  483481 Feb 24 19:02 indonesian-sentences.txt
+```
+
+#### Using `lm_optimizer.py` with the text file to generate values for the parameters `--default_alpha` and `--default_beta` that are used by the `lm_optimizer.py` script
+
+The `lm_optimizer.py` script is located in the `DeepSpeech` directory if you have set up your [environment][ENVIRONMENT.md] as outlined in the PlayBook.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# ls | grep lm_optimizer.py
+lm_optimizer.py
+```
+
+This script takes a text file, e.g. `indonesian-sentences.txt`, and a set of _checkpoints_, and determines the optimal `--default_alpha` and `--default_beta` values. These are then passed to `generate_lm.py` in the next step.
+
+**This step assumes you already have a trained model, and a set of saved _checkpoints_. For more information on _checkpoints_, see the [training](TRAINING.md) page.**
+
+Call `lm_optimizer.py` and pass it the text file, `--train_files`, `--dev_files`, `--test_files` and a `--checkpoint_dir` directory.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# python3 lm_optimizer.py \
+>   deepspeech-data/indonesian-sentences.txt \
+>   --train_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/train.csv \
+>   --dev_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/dev.csv \
+>   --test_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/test.csv \
```

Comment:
This is the only one considered by the code, so you don't need `--train_files` or `--dev_files`

Reply:
Thx

Reply:
Thx